src/reanimated2/screenTransition/swipeSimulator.ts

```diff
@@ -0,0 +1,228 @@
+  isTransitionCanceled: boolean
+) {
+  'worklet';
+  const screenDimensions = screenTransitionConfig.screenDimensions;
+  const progressX = Math.abs(event.translationX / screenDimensions.width);
+  const progressY = Math.abs(event.translationY / screenDimensions.height);
+  const progress = isTransitionCanceled
+    ? Math.max(progressX, progressY) / 2
+    : Math.max(progressX, progressY);
+  return progress;
+}
+
+function maybeScheduleNextFrame(
+  step: () => void,
+  didScreenReachDestination: boolean,
+  screenTransitionConfig: ScreenTransitionConfig,
+  event: PanGestureHandlerEventPayload,
+  isTransitionCanceled: boolean
+) {
+  'worklet';
+  if (!didScreenReachDestination) {
+    const stackTag = screenTransitionConfig.stackTag;
+    const progress = computeProgress(
+      screenTransitionConfig,
+      event,
+      isTransitionCanceled
+    );
+    RNScreensTurboModule.updateTransition(stackTag, progress);
+    requestAnimationFrame(step);
+  } else if (screenTransitionConfig.onFinishAnimation) {
+    screenTransitionConfig.onFinishAnimation();
+  }
+}
+
+export function swipeSimulator(
+  event: PanGestureHandlerEventPayload,
+  screenTransitionConfig: ScreenTransitionConfig,
+  lockAxis?: 'x' | 'y' | undefined
+) {
+  'worklet';
+  const screenSize = screenTransitionConfig.screenDimensions;
+  const startTimestamp = _getAnimationTimestamp();
+  const isTransitionCanceled = screenTransitionConfig.isTransitionCanceled;
+  const startingPosition = {
+    x: event.translationX,
+    y: event.translationY,
+  };
+  const direction = {
+    x: Math.sign(event.translationX),
+    y: Math.sign(event.translationY),
+  };
+  const finalPosition = isTransitionCanceled
+    ? { x: 0, y: 0 }
+    : { x: direction.x * screenSize.width, y: direction.y * screenSize.height };
+  const distance = {
+    x: Math.abs(finalPosition.x - startingPosition.x),
+    y: Math.abs(finalPosition.y - startingPosition.y),
+  };
+  const didScreenReachDestination = {
+    x: false,
+    y: false,
+  };
+  const BASE_VELOCITY = 300;
+  const velocity = { x: BASE_VELOCITY, y: BASE_VELOCITY };
+  if (lockAxis === 'x') {
+    velocity.y = 0;
+  } else if (lockAxis === 'y') {
+    velocity.x = 0;
+  } else {
+    if (Math.abs(startingPosition.x) > Math.abs(startingPosition.y)) {
+      velocity.x = BASE_VELOCITY;
+      velocity.y =
+        BASE_VELOCITY * Math.abs(startingPosition.y / startingPosition.x);
+    } else {
+      velocity.x =
+        BASE_VELOCITY * Math.abs(startingPosition.x / startingPosition.y);
+      velocity.y = BASE_VELOCITY;
+    }
+  }
+
+  if (isTransitionCanceled) {
+    const didScreenReachDestinationCheck = () => {
+      if (!lockAxis) {
+        return didScreenReachDestination.x && didScreenReachDestination.y;
+      } else if (lockAxis === 'x') {
+        return didScreenReachDestination.x;
+      } else {
+        return didScreenReachDestination.y;
+      }
+    };
+    const computeFrame = () => {
```

Comment:
Let's use function declarations instead of arrow functions for better stack traces if something goes wrong (arrow functions are anonymous).
```suggestion
    function computeFrame() {
```